src/utility-types.ts

```diff
@@ -1,6 +1,12 @@
 export type Draft<T> = { -readonly [P in keyof T]?: T[P] }
 
-export interface MessageValue<Data> {
+export type JSONPrimitive = number | boolean | string | null
+// eslint-disable-next-line no-use-before-define
+export type JSONValue = JSONPrimitive | JSONArray | JSONObject
+export type JSONObject = { [k: string]: JSONValue }
```

Comment:
The type shortcut is better: Record<string, JSONValue>

Reply:
This is NOT possible, because with that you get a reference loop